src/components/application_manager/src/rpc_service_impl.cc

```diff
@@ -245,6 +246,12 @@ bool RPCServiceImpl::ManageMobileCommand(
           app->hmi_level(mobile_apis::PredefinedWindows::DEFAULT_WINDOW);
     }
 
+    commands::CommandRequestImpl* request =
+        dynamic_cast<commands::CommandRequestImpl*>(command.get());
+    if (request != NULL) {
+      request->set_warning_info(warning_info);
```

Comment:
Can we move set_warning_info to base class?

Reply:
I can't see it being useful for any other type of message, but I do agree that casting it down is a bit ugly. I can move it to make things a bit more clean.

Reply:
Fixed in https://github.com/smartdevicelink/sdl_core/pull/3385/commits/6840997e72ef482f670c747b72ac78ecc8d29fb3